src/test/java/tdl/datapoint/coverage/CoverageDatapointAcceptanceTest.java

```diff
@@ -138,8 +140,12 @@ private String wrapAsSNSEvent(S3Event s3Event) throws JsonProcessingException {
 
     //~~~~~~~~~~ Helpers ~~~~~~~~~~~~~`
 
-    private static void waitForQueueToReceiveEvents() throws InterruptedException {
-        Thread.sleep(10000);
+    private void waitForQueueToReceiveEvents() throws InterruptedException {
+        int retryCtr = 0;
+        while ((coverageComputedEvents.size() < 2) && (retryCtr < TASK_FINISH_CHECK_RETRY_COUNT)) {
```

Comment:
Is there a better way, like asking the queue to trigger an event when events have been received, and we can check for count of the events received instead of what I;m doing?

Reply:
`sqsEventQueue.size()`?

Reply:
If absent we should write it driven by this test.